asapdiscovery-docking/asapdiscovery/docking/docking_v2.py

```diff
@@ -0,0 +1,283 @@
+import abc
+from enum import Enum
+from pathlib import Path
+from typing import Literal, Optional, Union
+
+import dask
+from asapdiscovery.data.dask_utils import actualise_dask_delayed_iterable
+from asapdiscovery.data.openeye import (
+    combine_protein_ligand,
+    oechem,
+    oedocking,
+    oeomega,
+    save_openeye_pdb,
+)
+from asapdiscovery.data.schema_v2.ligand import Ligand, compound_names_unique
+from asapdiscovery.data.schema_v2.pairs import DockingInputPair
+from asapdiscovery.docking.docking_data_validation import DockingResultCols
+from asapdiscovery.modeling.modeling import split_openeye_design_unit
+from pydantic import BaseModel, Field, PositiveFloat, PositiveInt, root_validator
+
+
+class SCORE_TYPE(Enum):
+    CHEMGAUSS4 = "chemgauss4"
+
+
+class DockingResult(BaseModel):
+    """
+    Schema for a DockingResult, containing both a DockingInputPair used as input to the workflow
+    and a Ligand object containing the docked pose.
+    Also contains the probability and chemgauss4 score of the docked pose.
+    """
+
+    input_pair: DockingInputPair = Field(description="Input pair")
+    posed_ligand: Ligand = Field(description="Posed ligand")
+    probability: Optional[PositiveFloat] = Field(description="Probability")
+    score_type: SCORE_TYPE = Field(description="Docking score type")
+    score: float = Field(description="Docking score")
+    provenance: dict[str, str] = Field(description="Provenance")
```

Comment:
We could have a settings field where just dict the docking method used to preserve the settings as well.

Reply:
I am going to go back through and sort out all the provenance.